extensions/arc/src/ui/dashboards/miaa/miaaComputeAndStoragePage.ts

```diff
@@ -0,0 +1,371 @@
+			this.createConfigurationSectionContainer(loc.memoryRequest, this.memoryRequestBox!),
+			this.createConfigurationSectionContainer(loc.memoryLimit, this.memoryLimitBox!)
+
+		];
+	}
+
+	private createConfigurationSectionContainer(key: string, input: azdata.Component): azdata.FlexContainer {
+		const inputFlex = { flex: '0 1 150px' };
+		const keyFlex = { flex: `0 1 250px` };
+
+		const flexContainer = this.modelView.modelBuilder.flexContainer().withLayout({
+			flexWrap: 'wrap',
+			alignItems: 'center'
+		}).component();
+
+		const keyComponent = this.modelView.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+			value: key,
+			CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px' }
+		}).component();
+
+		const keyContainer = this.modelView.modelBuilder.flexContainer().withLayout({ alignItems: 'center' }).component();
+		keyContainer.addItem(keyComponent, { CSSStyles: { 'margin-right': '0px', 'margin-bottom': '15px' } });
+		flexContainer.addItem(keyContainer, keyFlex);
+
+		const inputContainer = this.modelView.modelBuilder.flexContainer().withLayout({ alignItems: 'center' }).component();
+		inputContainer.addItem(input, { CSSStyles: { 'margin-bottom': '15px', 'min-width': '50px', 'max-width': '225px' } });
+
+		flexContainer.addItem(inputContainer, inputFlex);
+
+		return flexContainer;
+	}
+
+	private handleOnTextChanged(component: azdata.InputBoxComponent): boolean {
+		if ((!component.value)) {
+			// if there is no text found in the inputbox component return false
+			return false;
+		} else if ((!component.valid)) {
+			// if value given by user is not valid enable discard button for user
+			// to clear all inputs and return false
+			this.discardButton!.enabled = true;
+			return false;
+		} else {
+			// if a valid value has been entered into the input box, enable save and discard buttons
+			// so that user could choose to either edit instance or clear all inputs
+			// return true
+			this.saveButton!.enabled = true;
+			this.discardButton!.enabled = true;
+			return true;
+		}
+
+	}
+
+	private editCores(): void {
+		let currentCPUSize = this._miaaModel.config?.spec?.requests?.vcores;
+
+		if (!currentCPUSize) {
+			currentCPUSize = '';
+		}
+
+		this.coresRequestBox!.placeHolder = currentCPUSize;
+		this.coresRequestBox!.value = '';
+		this.saveArgs.coresRequest = undefined;
+
+		currentCPUSize = this._miaaModel.config?.spec?.limits?.vcores;
+
+		if (!currentCPUSize) {
+			currentCPUSize = '';
+		}
+
+		this.coresLimitBox!.placeHolder = currentCPUSize;
+		this.coresLimitBox!.value = '';
+		this.saveArgs.coresLimit = undefined;
+	}
+
+	private editMemory(): void {
+		let currentMemSizeConversion: string;
+		let currentMemorySize = this._miaaModel.config?.spec?.requests?.memory;
+
+		if (!currentMemorySize) {
+			currentMemSizeConversion = '';
+		} else {
+			currentMemSizeConversion = convertToGibibyteString(currentMemorySize);
```

Comment:
Would be good to test this conversion by deploying sqlmi with different memory units (e.g. `-ml 2098000Ki`, `-ml 2048Mi`, `-ml 2Gi`) and confirm they get converted correctly.  It looks like the formatting is the same between pg and sqlmi, so I expect it to work.

Reply:
better yet, add tests for those values.

Reply:
These already exist : https://github.com/microsoft/azuredatastudio/blob/39b6cc193ff51d0ce260df35496017b7b9b474eb/extensions/arc/src/test/common/utils.test.ts#L258

He's saying verify that azdata is giving us the expected format for the units for MIAA (these functions were developed for PG but should be the same for MIAA)

Reply:
Validated that we do convert them correctly